lib/generator.js

```diff
@@ -1429,25 +1450,23 @@
      */
     Generator.prototype.savePixmap = function (pixmap, path, settings) {
         var convert = require("./convert");
-
-        if (settings._scale) {
-            settings._scale = parseFloat(settings._scale);
-        }
-
-        // check that arguments are of the correct type
-        pixmap.width          = parseInt(pixmap.width, 10);
-        pixmap.height         = parseInt(pixmap.height, 10);
-        pixmap.bitsPerChannel = parseInt(pixmap.bitsPerChannel, 10);
-        if (settings.hasOwnProperty("quality")) {
-            settings.quality  = parseInt(settings.quality, 10);
-        }
-        if (settings.hasOwnProperty("ppi")) {
-            settings.ppi      = parseFloat(settings.ppi);
-        }
-
+        this._parsePixmapProperties(pixmap);
+        this._parsePixmapSaveSettings(settings);
         return convert.savePixmap(this._paths, pixmap, path, settings);
     };
-    
+
+    /**
+     * @param {!Pixmap}  pixmap                An object representing the layer's image. See savePixmap.
+     * @param {!Stream}  outputStream          A Stream object to receive the converted pixmap.
+     * @param {!Object}  settings              An object with settings for converting the image. See savePixmap.
+     * @outputStream
```

Comment:
This isn't correct JSDoc. I think `@outputStream` is spurious. This should have an `@return ...`